### PR TITLE
feat(TFD-1673): convert the ES input String into avro record

### PR DESCRIPTION
**Pull request by StefanLMNG**
**What is the current behavior?** (You can also link to an open issue here)

Elasticsearch input was mapped as string

**What is the new behavior?**

Elasticsearch input is mapped as record

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1076/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1076/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1077/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1077/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1078/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1078/)

File: components/components-elasticsearch/elasticsearch-runtime_2_4/src/main/java/org/talend/components/elasticsearch/runtime_2_4/ElasticsearchInputRuntime.java [hunk 1]
```diff
@@ -80,8 +89,19 @@ public ValidationResult initialize(RuntimeContainer container, ElasticsearchInpu
...
+                if (jsonGenericRecordConverter == null) {
+                    JsonSchemaInferrer jsonSchemaInferrer = new JsonSchemaInferrer(new ObjectMapper());
+                    Schema jsonSchema = jsonSchemaInferrer.inferSchema(c.element().toString());
+                    jsonGenericRecordConverter = new JsonGenericRecordConverter(jsonSchema);
+                }
+                GenericRecord outputRecord = jsonGenericRecordConverter.convertToAvro(c.element().toString());
+                c.output(outputRecord);
+            }
+        }));
+        return elasticsearchDataAsAvro.apply(ConvertToIndexedRecord.<GenericRecord>of());
```
Comment (RyanSkraba) [inline components/components-elasticsearch/elasticsearch-runtime_2_4/src/main/java/org/talend/components/elasticsearch/runtime_2_4/ElasticsearchInputRuntime.java]: `ConvertToIndexedRecord.<GenericRecord>of()` is unnecessary if you're already a PCollection of ? extends IndexedRecord! 

If you get java generics errors, you can make `PCollection<IndexedRecord> elasticsearchDataAsAvro`

File: components/components-elasticsearch/elasticsearch-runtime_2_4/src/test/java/org/talend/components/elasticsearch/runtime_2_4/ElasticsearchBeamRuntimeTestIT.java [hunk 2]
```diff
@@ -119,18 +124,68 @@ public void getSampleTest() {
 
         ElasticsearchDatasetRuntime datasetRuntime = new ElasticsearchDatasetRuntime();
         datasetRuntime.initialize(null, datasetProperties);
-        final List<String> samples = new ArrayList<>();
+        final List<IndexedRecord> samples = new ArrayList<>();
         datasetRuntime.getSample(3, new Consumer<IndexedRecord>() {
 
             ExtractJson parse = new ExtractJson();
```
Comment (RyanSkraba) [inline components/components-elasticsearch/elasticsearch-runtime_2_4/src/test/java/org/talend/components/elasticsearch/runtime_2_4/ElasticsearchBeamRuntimeTestIT.java]: Delete me.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1093/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1093/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1094/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1094/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aslemoing%2FTFD-1673_convert_json_into_avro_ES)

**Pull request merged**